processing/src/main/java/org/apache/druid/frame/processor/SuperSorter.java

```diff
@@ -517,19 +504,26 @@ private boolean runNextMiddleMerger()
               (pos == maxChannelsPerProcessor - 1 || (totalInputs != UNKNOWN_TOTAL && w == totalInputs - 1))) {
             // We found a set to merge. Let's collect the input channels and launch the merger.
             final List<ReadableFrameChannel> in = new ArrayList<>();
+            final List<PartitionedReadableFrameChannel> partitionedReadableFrameChannels = new ArrayList<>();
             for (long i = currentSetStart; i < currentSetStart + maxChannelsPerProcessor; i++) {
               if (inputsReady.remove(i)) {
-                try {
-                  final FrameFile handle = FrameFile.open(mergerOutputFile(inLevel, i), FrameFile.Flag.DELETE_ON_CLOSE);
-                  in.add(new ReadableFileFrameChannel(handle));
-                }
-                catch (IOException e) {
-                  throw new RuntimeException(e);
-                }
+                String levelAndRankKey = StringUtils.format("merged.%d.%d", inLevel, i);
```

Comment:
`StringUtils.format("merged.%d.%d", inLevel, i);` is used somewhere else as well when updating the map. I think it would be better to extract it out in a function analogous to the original `mergerOutputFile()`

Reply:
done, thanks!